openwisp_notifications/handlers.py

```diff
@@ -182,7 +183,12 @@ def send_email_notification(sender, instance, created, **kwargs):
         # therefore send email anyway.
         email_preference = True
 
-    if not (email_preference and instance.recipient.email):
+    email_address = EmailAddress.objects.filter(
+        user=instance.recipient, email=instance.recipient.email
+    ).first()
+    is_email_verified = email_address.verified if email_address else False
```

Comment:
I think this can be shortened to just:

```
email_verified = recipient.emailaddress_set.filter(verified=True, email=recipient.email).exists()
```

Reply:
Updated with the changes.